.github/workflows/build-and-push.yaml

```diff
@@ -50,10 +54,28 @@ jobs:
       run: |
         make build-image
 
+    - name: Write signing key to disk (only needed for `cosign sign --key`)
```

Comment:
As discussed, add `if: ${{ startsWith(github.ref, 'refs/tags/') }}` bellow `-name` to sign only tags